File: src/main/java/com/hwscs/backend/service/interfaces/DashboardService.java
package com.hwscs.backend.service.interfaces;

import com.hwscs.backend.dto.response.DutyOfficerDashboardDto;
import com.hwscs.backend.dto.response.InchargeDashboardDto;
import com.hwscs.backend.dto.response.NurseDashboardDto;

public interface DashboardService {
    NurseDashboardDto getNurseDashboard(String username);

    InchargeDashboardDto getInchargeDashboard(String username);

    DutyOfficerDashboardDto getDutyOfficerDashboard(String username);
}

File: frontend/src/app/core/services/dashboard.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { DutyOfficerDashboardDto, InchargeDashboardDto, NurseDashboardDto } from '../models/models';

@Injectable({
  providedIn: 'root'
})
export class DashboardService {
  private readonly apiUrl = '/api/dashboard';

  constructor(private http: HttpClient) {}

  getNurseDashboard(): Observable<NurseDashboardDto> {
    return this.http.get<NurseDashboardDto>(`${this.apiUrl}/nurse`);
  }

  getInchargeDashboard(): Observable<InchargeDashboardDto> {
    return this.http.get<InchargeDashboardDto>(`${this.apiUrl}/incharge`);
  }

  getDutyOfficerDashboard(): Observable<DutyOfficerDashboardDto> {
    return this.http.get<DutyOfficerDashboardDto>(`${this.apiUrl}/duty-officer`);
  }
}
